File: src/main/java/de/tectoast/emolga/commands/rpi/StartServerCommand.java
package de.tectoast.emolga.commands.rpi;

import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.emolga.commands.RPICommand;
import de.tectoast.emolga.utils.GPIOManager;

public class StartServerCommand extends RPICommand {


    public StartServerCommand() {
        super("startserver", "Startet den Server");
    }

    @Override
    public void process(GuildCommandEvent e) throws Exception {
        GPIOManager.startServer();
        e.reply("Der Server wurde gestartet!");
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/various/StartServerCommand.kt
package de.tectoast.emolga.commands.various

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent
import de.tectoast.emolga.utils.Constants
import de.tectoast.emolga.utils.GPIOManager
import de.tectoast.emolga.utils.PC

class StartServerCommand : Command(
    "startserver",
    "Startet den Server (und damit Terraria)",
    CommandCategory.Various,
    Constants.G.GENSHINEMPIRES
) {

    init {
        argumentTemplate = ArgumentManagerTemplate.noArgs()
    }

    override suspend fun process(e: GuildCommandEvent) {
        val pc = PC.FLORIX_2
        val gpio = GPIOManager()
        if (gpio.isOn(pc)) {
            e.reply("Der Server ist bereits an!")
            return
        }
        gpio.startServer(pc)
        e.reply("Der Server-PC wurde gestartet, der Terraria-Server dürfte demnächst online sein c:")
    }

}
